quinn/src/udp.rs

```diff
@@ -53,14 +49,9 @@ impl UdpSocket {
         meta: &mut [RecvMeta],
     ) -> Poll<io::Result<usize>> {
         debug_assert!(!bufs.is_empty());
-        ready!(self.io.poll_read_ready(cx, mio::Ready::readable()))?;
-        match self.io.get_ref().recv_ext(bufs, meta) {
-            Ok(n) => Poll::Ready(Ok(n)),
-            Err(ref e) if e.kind() == io::ErrorKind::WouldBlock => {
-                self.io.clear_read_ready(cx, mio::Ready::readable())?;
-                Poll::Pending
-            }
-            Err(e) => Poll::Ready(Err(e)),
+        match ready!(self.io.poll_read_ready(cx))?.try_io(|io| io.get_ref().recv_ext(bufs, meta)) {
+            Ok(res) => Poll::Ready(res),
+            Err(_) => Poll::Pending,
         }
```

Comment:
This is also wrong. You should loop when it returns an error, like in the example I linked.

Reply:
Ah, that's surprising -- thanks for noticing -- should have focused more on the example as you suggested.